### PR TITLE
Bump LLVM to 5e5a22caf88ac1ccfa8dc5720295fdeba0ad9372 

### PR DESCRIPTION
**Pull request by flemairen6**
Bump LLVM to 5e5a22caf88ac1ccfa8dc5720295fdeba0ad9372 and stableHlo to 83f095e7217c897f1eccac5652600ceb944cb0e0 to align with https://github.com/llvm/torch-mlir/pull/2598 
The typed pointers seems to have been deprecated for good, so I removed stuff that was related to them. I might have missed some.
I also see some tests from `krnl_to_llvm` failing as if they could not lower some operators. I have really little experience with this area of the project, it would be great if someone could have a look at the issue with me.

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
@jenkins-droid test this please

**Comment by AlexandreEichenberger** [comment]
(note it fails for Mac and Window too).

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
@flemairen6 On Linux Jenkins, I saw this kind of error when building onnx-mlir:
```
/workdir/onnx-mlir/src/Accelerators/NNPA/Transform/ZLow/ZLowRewrite.cpp:370:32: error: call to 'getPermutationMap' is ambiguous
        AffineMap permuteMap = AffineMap::getPermutationMap({0, 2, 3, 1}, ctx);
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/workdir/llvm-project/mlir/include/mlir/IR/AffineMap.h:104:20: note: candidate function
  static AffineMap getPermutationMap(ArrayRef<unsigned> permutation,
```

Perhaps, you should fix it first.

File: test/mlir/conversion/krnl_to_llvm/entry_point.mlir [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: onnx-mlir-opt --convert-krnl-to-llvm="use-opaque-pointers=true" --canonicalize %s -split-input-file | FileCheck %s
+// RUN: onnx-mlir-opt --canonicalize %s -split-input-file | FileCheck %s
```
Comment (tungld) [inline test/mlir/conversion/krnl_to_llvm/entry_point.mlir:1]: We have to keep `--convert-krnl-to-llvm` to lower krnl to llvm. So removing `="use-opaque-pointers=true"` should be enough.

File: test/mlir/conversion/krnl_to_llvm/krnl_C_library_function_lowering.mlir [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: onnx-mlir-opt -O3 --convert-krnl-to-affine --convert-krnl-to-llvm="use-opaque-pointers=true" %s -split-input-file | FileCheck %s
+// RUN: onnx-mlir-opt -O3 --convert-krnl-to-affine %s -split-input-file | FileCheck %s
```
Comment (tungld) [inline test/mlir/conversion/krnl_to_llvm/krnl_C_library_function_lowering.mlir:1]: Same here, we want to keep `--convert-krnl-to-llvm`

**Review by tungld — COMMENTED** [review]
In many lit tests, I see you remove `--convert-krnl-to-llvm="use-opaque-pointers=true"` completely. We should remove `="use-opaque-pointers=true"` only and keep `--convert-krnl-to-llvm`

File: test/mlir/conversion/krnl_to_llvm/entry_point.mlir (hunk 1 above)
Comment (flemairen6) [reply test/mlir/conversion/krnl_to_llvm/entry_point.mlir:1]: How did I not notice 🤦 Thank you very much for checking, I'll fix that right away

**Comment by flemairen6** [comment]
> @flemairen6 On Linux Jenkins, I saw this kind of error when building onnx-mlir:
> 
> ```
> /workdir/onnx-mlir/src/Accelerators/NNPA/Transform/ZLow/ZLowRewrite.cpp:370:32: error: call to 'getPermutationMap' is ambiguous
>         AffineMap permuteMap = AffineMap::getPermutationMap({0, 2, 3, 1}, ctx);
>                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /workdir/llvm-project/mlir/include/mlir/IR/AffineMap.h:104:20: note: candidate function
>   static AffineMap getPermutationMap(ArrayRef<unsigned> permutation,
> ```
> 
> Perhaps, you should fix it first.

I was not able to reproduce this error in the docker, but I still "fixed" the issue I was seeing in the pipeline.
I also digged a little and tried to fix the tests for krnl-to-llvm (the ones I hadn't break myself)

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by AlexandreEichenberger** [comment]
@flemairen6 looking at a few test cases that are failing to give you a hand.

**Comment by AlexandreEichenberger** [comment]
Because I did not want to push changes in your branch, I updated your PR with the latest of main, and then started to fix a few lit tests

Feel free to re-integrate my changes into yours, or work from mine, either way is fine

#2657

**Comment by tungld** [comment]
@flemairen6 , @AlexandreEichenberger I created PR #2658 to fix the remaining lit tests based your PRs

**Comment by tungld** [comment]
> Thank you very much, should we close this PR then? 

Yes, please, I will let you the author of the new PR when merging it since you did the most important part.

**Comment by flemairen6** [comment]
Closed in favor of https://github.com/onnx/onnx-mlir/pull/2658

**Pull request closed without merging**